packages/core/src/components/essentials/tooltip-bar.ts

```diff
@@ -0,0 +1,201 @@
+		const chartprefix = Tools.getProperty(this.model.getOptions(), "style", "prefix");
+		this.tooltip = DOMUtils.appendOrSelect(holder, `div.${settings.prefix}--${chartprefix}--tooltip`);
+
+		// Apply html content to the tooltip
+		const tooltipTextContainer = DOMUtils.appendOrSelect(this.tooltip, "div.content-box");
+
+		// listen to show-tooltip Custom Events to render the tooltip
+		this.services.events.getDocumentFragment().addEventListener("show-tooltip", e => {
+			// check the type of tooltip and that it is enabled
+			if ((e.detail.type === TooltipTypes.DATAPOINT && Tools.getProperty(this.model.getOptions(), "tooltip", "datapoint", "enabled"))
+				|| (e.detail.type === TooltipTypes.GRIDLINE && Tools.getProperty(this.model.getOptions(), "tooltip", "gridline", "enabled")) ) {
+
+				const hoveredElement = e.detail.hoveredElement.node();
+
+				// if there is a provided tooltip HTML function
+				if (Tools.getProperty(this.model.getOptions(), "tooltip", "customHTML")) {
+					tooltipTextContainer.html(this.model.getOptions().tooltip.customHTML(hoveredElement));
+				} else {
+					if (e.detail.multidata) {
+						// multi tooltip
+						tooltipTextContainer.html(this.getMultilineTooltipHTML(e.detail.multidata));
+						// Position the tooltip
+						this.positionTooltip();
+					} else {
+						const data = e.detail.hoveredElement.datum();
+						tooltipTextContainer.html(this.getTooltipHTML(data));
+
+						const position = this.getTooltipPosition(hoveredElement);
+
+						// Position the tooltip relative to the bars
+						this.positionTooltip(position);
+					}
+				}
+				// Fade in
+				this.tooltip.classed("hidden", false);
+			}
+		});
+
+		// listen to hide-tooltip Custom Events to hide the tooltip
+		this.services.events.getDocumentFragment().addEventListener("hide-tooltip", () => {
+			this.tooltip.classed("hidden", true);
+		});
+	}
+
+	/**
+	 * Get the position of the tooltip relative to the active hovered bar. Tooltip should appear above
+	 * positive valued data and below negative value data.
+	 * @param hoveredElement
+	 */
+	getTooltipPosition(hoveredElement) {
+		const data = select(hoveredElement).datum() as any;
+
+		const holderPosition = select(this.services.domUtils.getHolder()).node().getBoundingClientRect();
+		const barPosition = hoveredElement.getBoundingClientRect();
+
+		const { verticalOffset } = this.model.getOptions().tooltip.datapoint;
+		// if there is a negative value bar chart, need to place the tooltip below the bar
+		if (data.value <= 0) {
+			// negative bars
+			const tooltipPos = {
+				left: (barPosition.left - holderPosition.left) + barPosition.width / 2,
+				top: (barPosition.bottom - holderPosition.top) + verticalOffset
+			};
+
+			return {placement: TooltipPosition.BOTTOM, position: tooltipPos};
+		} else {
+			// positive bars
+			const tooltipPos = {
+				left: (barPosition.left - holderPosition.left) + barPosition.width / 2,
+				top: (barPosition.top - holderPosition.top) - verticalOffset
```

Comment:
it differs in how it calculates the placements also, so if there is a negative bar, it uses the **barPosition.bottom** and positive bars use **barPosition.top** when calculating the `tooltipPos.top` . Also, it returns different placements: positive bars get the tooltip **tooltipPosition.TOP**, and negative gets **tooltipPosition.BOTTOM**. 

I'll look at optimizing it but i originally thought this would be better for readability